ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-commands.sh

```diff
@@ -7,20 +7,40 @@ set -o pipefail
 export PATH=$PATH:/tmp/bin
 mkdir -p /tmp/bin
 
-export GITHUB_USER GITHUB_TOKEN QUAY_TOKEN QUAY_OAUTH_USER QUAY_OAUTH_TOKEN QUAY_OAUTH_TOKEN_RELEASE_SOURCE QUAY_OAUTH_TOKEN_RELEASE_DESTINATION OPENSHIFT_API OPENSHIFT_USERNAME OPENSHIFT_PASSWORD
+export GITHUB_USER GITHUB_TOKEN QUAY_TOKEN QUAY_OAUTH_USER QUAY_OAUTH_TOKEN QUAY_OAUTH_TOKEN_RELEASE_SOURCE QUAY_OAUTH_TOKEN_RELEASE_DESTINATION OPENSHIFT_API OPENSHIFT_USERNAME OPENSHIFT_PASSWORD \
+    GITHUB_ACCOUNTS_ARRAY PREVIOUS_RATE_REMAINING GITHUB_USERNAME_ARRAY GH_RATE_REMAINING
 
-GITHUB_USER=$(cat /usr/local/ci-secrets/redhat-appstudio-qe/github-user)
-GITHUB_TOKEN=$(cat /usr/local/ci-secrets/redhat-appstudio-qe/github-token)
 QUAY_TOKEN=$(cat /usr/local/ci-secrets/redhat-appstudio-qe/quay-token)
 QUAY_OAUTH_USER=$(cat /usr/local/ci-secrets/redhat-appstudio-qe/quay-oauth-user)
 QUAY_OAUTH_TOKEN=$(cat /usr/local/ci-secrets/redhat-appstudio-qe/quay-oauth-token)
 QUAY_OAUTH_TOKEN_RELEASE_SOURCE=$(cat /usr/local/ci-secrets/redhat-appstudio-qe/quay-oauth-token-release-source)
 QUAY_OAUTH_TOKEN_RELEASE_DESTINATION=$(cat /usr/local/ci-secrets/redhat-appstudio-qe/quay-oauth-token-release-destination)
 OPENSHIFT_API="$(yq e '.clusters[0].cluster.server' $KUBECONFIG)"
 OPENSHIFT_USERNAME="kubeadmin"
+PREVIOUS_RATE_REMAINING=0
 
-yq -i 'del(.clusters[].cluster.certificate-authority-data) | .clusters[].cluster.insecure-skip-tls-verify=true' $KUBECONFIG
+# user stored: username:token,username:token
+IFS=',' read -r -a GITHUB_ACCOUNTS_ARRAY <<< "$(cat /usr/local/ci-secrets/redhat-appstudio-qe/github_accounts)"
+for account in "${GITHUB_ACCOUNTS_ARRAY[@]}"
+do :
+    IFS=':' read -r -a GITHUB_USERNAME_ARRAY <<< "$account"
+
+    GH_RATE_REMAINING=$(curl -s \
+    -H "Accept: application/vnd.github+json" \
+    -H "Authorization: Bearer ${GITHUB_USERNAME_ARRAY[1]}"\
+    https://api.github.com/rate_limit | jq ".rate.remaining")
+
+    if [[ "${GH_RATE_REMAINING}" -ge "${PREVIOUS_RATE_REMAINING}" ]];then
+        GITHUB_USER="${GITHUB_USERNAME_ARRAY[0]}"
+        GITHUB_TOKEN="${GITHUB_USERNAME_ARRAY[1]}"
+    fi
+    PREVIOUS_RATE_REMAINING="${GH_RATE_REMAINING}"
+    echo "[INFO ] user: $GITHUB_USER with rate limit remaining $GH_RATE_REMAINING"
+done
```

Comment:
shouldn't we have [separate tokens for HAS and e2e-tests](https://redhat-internal.slack.com/archives/C02FANRBZQD/p1676473620886019?thread_ts=1676472720.902939&cid=C02FANRBZQD)?
i.e.
`HAS_GITHUB_TOKEN` - would be used for bootstrapping the cluster (equivalent to `MY_GITHUB_TOKEN`)
`GITHUB_TOKEN` - used by e2e tests for sending requests to Github API

Reply:
HAS do more requests to github than our tests... TBH i think will be better to have 3 users and balance the request between e2e and has